content/docs/user-guide/hydra-integration.md

```diff
@@ -0,0 +1,251 @@
+# Hydra Integration
+
+_New in DVC 2.25.0 (see `dvc version`)_
+
+In this guide we explain how to enable and use the DVC and
+[Hydra](https://hydra.cc/) integration.
+
+## How it works
+
+On each `dvc exp run` call, DVC will use Hydra to **compose** a single
+configuration object and **dump** it to `params.yaml`. This will happen
+**before** the experiment starts running.
+
+Instead of relying on the
+[`@hydra.main`](https://hydra.cc/docs/tutorials/basic/your_first_app/simple_cli/)
+Python decorator for composing the config object, we can now parse a single
+file: `params.yaml`.
+
+This allows to combine Hydra composition and DVC <abbr>parameters</abbr> in
+order to configure DVC <abbr>pipelines</abbr> (running multiple steps of
+different shell commands, instead of a single Python script) and use features
+like [Templating](/doc/user-guide/project-structure/dvcyaml-files#templating)
+and [`foreach`](/doc/user-guide/project-structure/dvcyaml-files#foreach).
+
+## Setting up Hydra
+
+<admon type="tip">
+
+Learn more about setting up Hydra in the
+[official Hydra tutorial](https://hydra.cc/docs/tutorials/basic/your_first_app/composition/).
+
+</admon>
+
+In order to start using Hydra composition capabilities it is required to have a
+directory of
+[Hydra Config Groups](https://hydra.cc/docs/tutorials/basic/your_first_app/config_groups/)
+(`conf`):
+
+```dvc
+$ tree conf
+conf
+├── config.yaml
+├── dataset
+│   ├── imagenette.yaml
+│   └── imagewoof.yaml
+└── train
+    ├── baseline.yaml
+    ├── model
+    │   ├── alexnet.yaml
+    │   ├── efficientnet.yaml
+    │   └── resnet.yaml
+    └── optimizer
+        ├── adam.yaml
+        └── sgd.yaml
+```
+
+Along with a file defining the top-level
+[Hydra Defaults List](https://hydra.cc/docs/tutorials/basic/your_first_app/defaults/)
+(`conf/config.yaml`):
+
+```dvc
+$ cat conf/config.yaml
+defaults:
+  - dataset: imagenette
+  - train: baseline
```

Comment:
Could you have the baseline defaults here to eliminate the nested defaults list?

Reply:
Yes, it is possible. 

I believe I was trying to build on the example repo something that mixed the basic pattern of single defaults with the more complex https://hydra.cc/docs/patterns/configuring_experiments/ 

But for this doc it makes more sense to keep it all in the top level defaults